types/service/elasticsearch_service_configuration.go

```diff
@@ -0,0 +1,95 @@
+package service
+
+import (
+	"errors"
+	"fmt"
+)
+
+const (
+	// ElasticsearchServiceTypeStandard is the elasticsearch
+	// service type for standard elasticsearch services.
+	ElasticsearchServiceTypeStandard = "standard"
+
+	// ElasticsearchAuthenticationTypeBasic is the basic authentication type for elasticsearch.
+	ElasticsearchAuthenticationTypeBasic = "basic"
+)
+
+// ElasticsearchServiceConfiguration represents service
+// configuration for elasticsearch services (fka sockets).
+type ElasticsearchServiceConfiguration struct {
+	ElasticsearchServiceType string `json:"elasticsearch_service_type"`
+
+	// mutually exclusive fields below
+	StandardElasticsearchServiceConfiguration *StandardElasticsearchServiceConfiguration `json:"standard_elasticsearch_service_configuration,omitempty"`
+}
+
+// StandardElasticsearchServiceConfiguration represents service
+// configuration for standard elasticsearch services (fka sockets).
+type StandardElasticsearchServiceConfiguration struct {
+	HostnameAndPort        // inherited
+	Protocol        string `json:"protocol"`
+	HostHeader      string `json:"host_header,omitempty"`
+
+	AuthenticationType  string                             `json:"authentication_type"`
+	BasicAuthentication *ElasticsearchServiceTypeBasicAuth `json:"basic_auth_configuration,omitempty"`
+}
+
+// ElasticsearchServiceTypeBasicAuth represents basic auth configuration that based on username and password.
+type ElasticsearchServiceTypeBasicAuth struct {
```

Comment:
do we need this additional level? or can we have the field BasicAuthentication be a UsernameAndPassword type?

Reply:
We may want to add some tls stuff later